cmd/kube-apiserver/app/server.go

```diff
@@ -339,7 +341,13 @@ func (s *APIServer) Run(_ []string) error {
 		glog.Fatalf("Invalid server address: %v", err)
 	}
 
-	etcdStorage, err := newEtcd(s.EtcdConfigFile, s.EtcdServerList, latest.GroupOrDie("").InterfacesFor, latest.GroupOrDie("").Version, s.StorageVersion, s.EtcdPathPrefix)
+	g, err := latest.Group("")
+	if err != nil {
+		return err
+	}
+	storageVersions := make(map[string]string)
```

Comment:
This is good-- we should have a map of group to storage version. (can we call it groupToStorageVersion?)

In the near future, maybe not this PR:

We should deprecate StorageVersion and ExpStorageVersion, and replace them with StorageVersions, which should be a map from group to version (it can be specified in the form: "experimental/v1alpha1,core/v1,/v1")

newEtcd should then be called in a loop, etc. Do we really need to call newEtcd for every group?